### PR TITLE
Milestone 3 Tutorial 2

### PR DESCRIPTION
**Pull request by dirkcuys**
This PR contains
- the tutorial for uploading a directory of files to IPFS
- an updated development environment bootstrap that can be used to create IPFS sites in Agregore Web
- a retrospective on the creation of the functionality and tutorial

### CODE REVIEW

**Review by RangerMauve — CHANGES_REQUESTED** [review]

**Comment by RangerMauve** [comment]
Mind moving the retrospective into the blog folder?

File: docs/tutorials/ipfs-dir-upload/index.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
+In this tutorial we will create an app that allows you to upload a whole directory of files to IPFS! The plan is to use this functionality in a future tutorial to add support for uploading 3rd-party dependencies, but you can use it for other purposes like uploading media to a site you want to build!
+
+## Getting started!
+
+We will be using a [self-hosted development environment](https://agregore.mauve.moe/docs/examples/browser-devenv/) for Agregore that we created in a [previous tutorial](https://agregore.mauve.moe/docs/tutorials/ipfs-browser-devenv/part-1).
```
[the remark this answers is not in the document]
Comment (dirkcuys) [reply docs/tutorials/ipfs-dir-upload/index.md]: makes sense, will do

File: docs/examples/browser-devenv-v2/files/style.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@import url("agregore://theme/vars.css");
+html {
+    font-family: var(--ag-theme-font-family);
```
[the remark this answers is not in the document]
Comment (dirkcuys) [reply docs/examples/browser-devenv-v2/files/style.css:3]: I think I did try that, but the styles for the editor didn't work well with the defaults and I haven't gotten around to fix that. If it's fine with you, I'd rather change this in the next tutorial

**Comment by dirkcuys** [comment]
I moved the retrospective and updated the structure a bit. I like what @tripledoublev did with it.

**Review by RangerMauve — APPROVED** [review]

**Pull request merged**